### ISSUE TITLE
Uncaught Error: TypeError: null is not an object (evaluating 'manifest.packagerOpts')

### ISSUE THREAD

**Issue by BrodaNoel** [expo/expo-cli#3934]
### Summary

I just upgraded to the last version (4.12.6) (released 8 hours ago), and I'm getting this while opening the app, in development time:
```
Blue screen:
Uncaught Error: TypeError: null is not an object (evaluating 'manifest.packagerOpts'), stack: loadModuleImplementation
```

Happening in real device, and emulator

### Environment

  Expo CLI 4.12.6 environment info:
    System:
      OS: macOS 11.6
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
      Yarn: 1.22.15 - ~/.yarn/bin/yarn
      npm: 7.20.3 - ~/.nvm/versions/node/v14.16.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 23, 25, 26, 27, 29, 30
        Build Tools: 23.0.1, 23.0.3, 25.0.3, 26.0.1, 27.0.3, 28.0.3, 29.0.2
        System Images: android-22 | Google APIs Intel x86 Atom, android-23 | Intel x86 Atom, android-23 | Google APIs Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6308749
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~42.0.1 => 42.0.3
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2
      react-native-web: ~0.13.12 => 0.13.18
    npmGlobalPackages:
      expo-cli: 4.12.6
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

Android Real Device, and emulator: Expo Go on Pixel_XL_Android_10, iPhone SE real device

### Error output

Uncaught Error: TypeError: null is not an object (evaluating 'manifest.packagerOpts'), stack: loadModuleImplementation

### Reproducible demo or steps to reproduce from a blank project

N/A

**Comment by BrodaNoel** [comment]
Seems like the problem is here:
```
import Constants from 'expo-constants';
const { manifest } = Constants;
```

Adding more info in few minutes

**Comment by BrodaNoel** [comment]
![image](https://user-images.githubusercontent.com/1837969/138196650-7cffc351-d033-49e9-be3c-f35b6384f569.png)

**Comment by BrodaNoel** [comment]
```
{
  "expo": {
    "name": "CNCor",
    "version": "1.5.0",
    "description": "App del Circulo de Navegantes de Cordoba",
    "slug": "asdasd",
    "orientation": "portrait",
    "primaryColor": "#ffffff",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0,
      "enabled": false
    },
    "assetBundlePatterns": [
      "assets/*.png",
      "assets/*.jpg",
      "components/**/*.png",
      "components/**/*.jpg",
      "pages/**/*.png",
      "pages/**/*.jpg",
      "pages/**/*.json",
      "UI/**/*.png",
      "UI/**/*.jpg",
      "UI/**/*.ttf"
    ],
    "ios": {
      "buildNumber": "1.5.0",
      "supportsTablet": true,
      "bundleIdentifier": "com.escueladeyachting.circulo",
      "googleServicesFile": "./GoogleService-Info.plist"
    },
    "android": {
      "versionCode": 7,
      "package": "com.escueladeyachting.circulo",
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "googleServicesFile": "./google-services.json"
    },
    "platforms": [
      "android",
      "ios"
    ],
    "extra": {
      "eas": {
        "projectId": "asdasd"
      }
    }
  }
}
```

**Comment by brentvatne** [comment]
i can't reproduce this on my end so there is likely something in your app that is leading to this occurring. i can't say really, you will need to share a reproducible example. try creating a new repository and reproducing it.

i did `expo init` and choose a blank app and then added the following to the top of the file:

```
import Constants from 'expo-constants';
const { manifest } = Constants;
console.log(manifest.packagerOpts);
```

this worked as expected

**Comment by brentvatne** [comment]
the manfiest is somehow null in your project. again i cannot say what is happening without a reproducible example. it's possible that you added some babel transform that broke it.

**Comment by BrodaNoel** [comment]
@brentvatne you know what is weird? I was coding yesterday, with no problems. Today I saw the "There is an update in expo-cli", so I ran `npm i -g expo-cli`, and then I got this error.
I'll try to get more info in a few

**Comment by brentvatne** [comment]
@BrodaNoel - then install the older version of expo-cli to verify ;)

**Comment by brentvatne** [comment]
thanks @BrodaNoel! we'll do a patch release to ensure this doesn't happen if you have the eas project id set in your app config

**Comment by brentvatne** [comment]
this is released in 4.12.7

**Comment by BrodaNoel** [comment]
Tested.
**Working**
